### ISSUE TITLE
d3.legendColor is not a function

### ISSUE THREAD

**Issue by reisner** [juba/scatterD3#68]
Hi there,

The plot works great for me, except for the legend. I have to set `legend_width = 0`, or else I get this Javascript error:

```
Uncaught TypeError: d3.legendColor is not a function
    at add_color_legend (http://127.0.0.1:3112/scatterD3-0.7/scatterD3-legend.js:24:27)
    at HTMLDivElement.<anonymous> (http://127.0.0.1:3112/scatterD3-binding-0.8.1/scatterD3.js:208:7)
    at pt.Jf [as each] (http://127.0.0.1:3112/d3-4.5.0/d3.min.js:4:27015)
    at chart (http://127.0.0.1:3112/scatterD3-binding-0.8.1/scatterD3.js:114:19)
    at pt.Vf [as call] (http://127.0.0.1:3112/d3-4.5.0/d3.min.js:4:26551)
    at Object.renderValue (http://127.0.0.1:3112/scatterD3-binding-0.8.1/scatterD3.js:800:5)
    at Object.renderValue (http://127.0.0.1:3112/htmlwidgets-1.0/htmlwidgets.js:819:25)
    at exports.OutputBinding.shinyBinding.renderValue (http://127.0.0.1:3112/htmlwidgets-1.0/htmlwidgets.js:516:20)
    at exports.OutputBinding.onValueChange (http://127.0.0.1:3112/shared/shiny.min.js:4:9871)
    at exports.OutputBinding.delegator.(anonymous function) [as onValueChange] (http://127.0.0.1:3112/htmlwidgets-1.0/htmlwidgets.js:112:23)
```

Any ideas about what I could be doing wrong?

**Comment by juba** [comment]
Hmmm this is strange. Would you have a reproducible example ? Are you using the latest development version ?

**Comment by reisner** [comment]
I'll try to put together a minimal example in the next couple days!

**Comment by reisner** [comment]
Ugh, that took a bit of digging. Looks like this package has some issues when used in conjunction with the `networkD3` package.

Here's the gist: https://gist.github.com/reisner/8611d10426d27a5ef9687d78f027ff93

**Comment by juba** [comment]
Thanks for the reproducible example. Are you using `scatterD3` 0.3.1 ? Because I believe this should be fixed in 0.3.2 (current CRAN version)...

**Comment by reisner** [comment]
I think you mean 0.8.1 and 0.8.2 -- Yeah you're right. I was using 0.8.1.

It works now, thanks!

**Comment by juba** [comment]
Yes, it was an issue regarding the compatibility between d3v3 and d3v4 widgets, which should be fixed since the 0.8.2 release. Thanks for taking the time to report it.
